File: src/main/java/hyeon9mak/multidatasourcequerycounter/QueryCountPerRequest.java
package hyeon9mak.multidatasourcequerycounter;

import org.springframework.stereotype.Component;
import org.springframework.web.context.annotation.RequestScope;

@RequestScope
@Component
public class QueryCountPerRequest {

    private String apiUrl;
    private int totalQueryCount;
    private long totalQueryMilliSeconds;

    public QueryCountPerRequest() {
        this.apiUrl = "";
        this.totalQueryCount = 0;
        this.totalQueryMilliSeconds = 0L;
    }

    public void incrementQueryCount(long executionMilliSeconds) {
        totalQueryCount++;
        totalQueryMilliSeconds += executionMilliSeconds;
    }

    public void updateApiUrl(String apiUrl) {
        this.apiUrl = apiUrl;
    }

    public String getApiUrl() {
        return apiUrl;
    }

    public int getTotalQueryCount() {
        return totalQueryCount;
    }

    public long getTotalQueryMilliSeconds() {
        return totalQueryMilliSeconds;
    }
}

File: src/main/java/hyeon9mak/multidatasourcequerycounter/QueryCountPerRequest.kt
package hyeon9mak.multidatasourcequerycounter

import java.util.concurrent.atomic.AtomicInteger
import java.util.concurrent.atomic.AtomicLong

data class QueryCountPerRequest(
    var apiUrl: String = "",
    private var _totalQueryCount: AtomicInteger = AtomicInteger(0),
    private var _totalQueryMilliSeconds: AtomicLong = AtomicLong(0L),
) {
    val totalQueryCount: Int
        get() = _totalQueryCount.get()

    val totalQueryMilliSeconds: Long
        get() = _totalQueryMilliSeconds.get()

    fun incrementQueryCount(executionMilliSeconds: Long) {
        _totalQueryCount.incrementAndGet()
        _totalQueryMilliSeconds.addAndGet(executionMilliSeconds)
    }
}
